File: projecteuler/exer09.ts
/*

The sum of the primes below 10 is 2 + 3 + 5 + 7 = 17.

Find the sum of all the primes below two million.

*/

export const isPrime = (n:number): boolean => {
  let div = 0;
  for (let i = 1; i <= n; i += 1) {
    if (n % i === 0) {
      div += 1;
    }
  }
  if (div === 2) return true;
  return false;
};

let sum = 0;

for (let i = 1; i < 2000000; i += 1) {
  if (isPrime(i)) {
    sum += i;
  }
}

console.log(sum);

File: projecteuler/exer09.cpp
#include<iostream>
using namespace std;

bool isPrime(int number) {
    int div = 0;
    for(int i = 1; i <= number; i++) {
        if(number % i == 0) {
            div++;
        }
    }
    if(div == 2) {
        return true;
    } else {
        return false;
    };
}

main() {
    int n = 100;
    bool b = isPrime(n);

    if(b) {
        cout<<"Primo";
    }
}